x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/extractor/scroll/ExtractedField.java

```diff
@@ -103,7 +103,13 @@ public static ExtractedField newField(String alias, String name, ExtractionMetho
             if (value.length != 1) {
                 return value;
             }
-            value[0] = ((BaseDateTime) value[0]).getMillis();
+            if (value[0] instanceof BaseDateTime) { // script field
+                value[0] = ((BaseDateTime) value[0]).getMillis();
+            } else if (value[0] instanceof String) { // doc_value field with the epoch_millis format
+                value[0] = Long.parseLong((String) value[0]);
```

Comment:
Is the value going to be a `String` rather than a `long` here since we're explicitly setting the format to `epoch_millis`?

Reply:
This is correct.